### ISSUE TITLE
Syncthing: Updated Version Uses Wrong Home Folder

### ISSUE THREAD

**Issue by MoisieSyncthing** [SynoCommunity/spksrc#3591]
Hello:

I've just updated my Synology Syncthing package from 0.14.30-13 -> 0.14.49-15. Since then, the Syncthing package appears to be ignoring my previously-configured Syncthing 'home' folder of `/usr/local/syncthing/var` - so it's running with a default set of settings.

Any clues on where I should correct this?

Thanks!

**Comment by acolomb** [comment]
How did you configure that home folder previously?

If you look at `/var/packages/syncthing/scripts/service-setup`, the `service_prestart()` function sets both `$HOME` and Syncthing's `-home` option to the path you actually intended. If not, please post the contents of that file from your NAS.

Under Syncthing's GUI settings, you can specify a different default path for new shared folders, but it sounds like that is not what you're after.

**Comment by BenjV** [comment]
You cannot set privileges for the user (sc-syncthing) of a package.
Instead you must use the group sc-syncthing to supply permissions to folder and files.
See:
https://github.com/SynoCommunity/spksrc/wiki/Permission-Management

**Comment by MoisieSyncthing** [comment]
> How did you configure that home folder previously?

I didn't do anything special - just used the default path provided by an earlier version of the Syncthing package.

> If you look at /var/packages/syncthing/scripts/service-setup, the service_prestart() function sets both $HOME and Syncthing's -home option to the path you actually intended. If not, please post the contents of that file from your NAS.

I've not changed anything in this respect in the service-setup script. The only thing I have done is to add another environment variable to limit the number of CPUs.

I presume there should be a link from in the package-specific location to `/usr/local/syncthing/var` to allow Syncthing to find this - should I go looking for this?

> Under Syncthing's GUI settings, you can specify a different default path for new shared folders, but it sounds like that is not what you're after.

That's right - it's not the default path, it's the location of the Syncthing home folder, containing the database, config.xml, etc. This is all currently in `/usr/local/syncthing/var` - but the log from Syncthing looks like it's trying to set up a default folder - so clearly isn't reading the settings.

> Or is it maybe just a permission problem for the aforementioned folder? You have to make sure that the sc-syncthing user has access to the folder, since that what the service runs as.

This is what the folder looks like:

`root@NAS02:/usr/local/syncthing# ls -lah`
`total 0`
`drwxr-xr-x 1 sc-syncthing syncthing  44 Jan 22 16:08 .`
`drwxr-xr-x 1 root         root      150 Jan 22 14:04 ..`
`drwxr-xr-x 1 sc-syncthing syncthing  24 Jul 30 22:23 app`
`drwxr-xr-x 1 sc-syncthing syncthing 172 Jan 22 16:08 bin`
`drwx------ 1 sc-syncthing syncthing 518 Jan 23 08:44 var`

> Your Syncthing logs (accessible from the Web GUI) could also point out if the configuration was not accessible, however then the service should have died already...

Yeah - can't get into the GUI at the moment either, possibly because it's defaulting to 127.0.0.1:8384, so I can't get in from an external browser...

Any thoughts much appreciated!

**Comment by MoisieSyncthing** [comment]
Hello!

Just wanted to flag up that I _might_ have resolved this now. I'll check on it, and post what happened as soon as I am able - just wanted to make sure no-one else wasted their time on it first though!

Thanks!

**Comment by acolomb** [comment]
Sound good, please do share your findings when ready.

**Comment by MoisieSyncthing** [comment]
Hello!

Ok - yes, it looks like I managed to fix this - and it was my own idiotic fault in the first place...

I had forgotten that I had tried to install the updated package some months ago, and then became a cropper because I was wanting some custom launch arguments - but couldn't work out how to implement them. So I downgraded the package back to the previous version, and carried on from there (whilst investigating the launch arguments on another system).

Once I'd upgraded the package again, my old custom launch arguments then prevented Syncthing from starting up properly, until I removed them. Specifically, I had added the following two lines to `/volume1/@appstore/syncthing/var/options.conf`:

`SYNCTHING_OPTIONS=“${SYNCTHING_OPTIONS} -logfile=/volume1/Logs/syncthing.log”
SYNCTHING_OPTIONS=“${SYNCTHING_OPTIONS} -verbose”`

Once I removed these, Syncthing launched and behaved as expected.

Thanks for your input and suggestions - and apologies for causing the noise!
